File: Python/66.Plus One.py
#返回一个数加一后的值(数可以任意大)
class Solution(object):
    def plusOne(self, digits):
        l = len(digits)
        if l==0:
            return []
        i = l-1
        flag = (digits[i]+1)//10
        digits[i] = (digits[i]+1)%10
        i = i-1
        while i>=0 and flag!=0:
            flag = (digits[i]+1)//10
            digits[i] = (digits[i]+1)%10
            i-=1
        if flag!=0:
            digits.insert(0,1)
        return digits

s = Solution()
print(s.plusOne([9,9,9,9]))

File: C++/Math/66.Plus One.cpp
/*
数组代表一个非负整数，求该数加1的结果。
*/

#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
	vector<int> plusOne(vector<int>& digits) {
        int flag = 1, sum=0;
        for(int i=digits.size()-1; i >=0; --i)
        {
        	sum = digits[i] + flag;
        	digits[i] = sum % 10;
        	flag = sum/10;
        }
        if(flag)	digits.insert(digits.begin(),1);
        return digits;
    }
};
int main()
{
	Solution s;
	vector<int> out{9,9,9};
	out = s.plusOne(out);
	for(int i:out)
		cout<<i<<" ";
	cout<<endl;
}

